native/cocos/renderer/gfx-base/GFXDef-common.h

```diff
@@ -1164,17 +1171,17 @@ struct RenderPassInfo {
     SubpassDependencyList  dependencies;
 };
 
-struct GlobalBarrierInfo {
-    std::vector<AccessType> prevAccesses;
-    std::vector<AccessType> nextAccesses;
+struct ALIGNAS(8) GeneralBarrierInfo {
+    AccessFlags prevAccesses{AccessFlagBit::NONE};
+    AccessFlags nextAccesses{AccessFlagBit::NONE};
 };
-using GlobalBarrierInfoList = vector<GlobalBarrierInfo>;
+using GeneralBarrierInfoList = vector<GeneralBarrierInfo>;
 
-struct TextureBarrierInfo {
-    std::vector<AccessType> prevAccesses;
-    std::vector<AccessType> nextAccesses;
+struct ALIGNAS(8) TextureBarrierInfo {
+    AccessFlags prevAccesses{AccessFlagBit::NONE};
+    AccessFlags nextAccesses{AccessFlagBit::NONE};
 
-    bool discardContents{false};
+    uint64_t discardContents{0}; // @ts-boolean
```

Comment:
Why not use bool as TS?

Reply:
The 64-bit type is used to make sure the memory layout is not implicitly padded.

Reply:
Got it, could you please add some comments for it?